Storage/SQL/SQLiteBookmarksModel.swift

```diff
@@ -855,4 +890,10 @@ public class MergedSQLiteBookmarks: BookmarksModelFactorySource {
         self.local = SQLiteBookmarks(db: db)
         self.buffer = SQLiteBookmarkBufferStorage(db: db)
     }
+
+    public func getURLForKeywordSearch(keyword: String) -> Deferred<Maybe<String>> {
+        return modelFactory >>== {
+            return ($0 as! KeywordSearchSource).getURLForKeywordSearch(keyword)
```

Comment:
If you make `SQLiteBookmarks` implement `KeywordSearchSource`, and put the logic there rather than in `SQLiteBookmarksModelFactory`, then this simply becomes `return self.local.getURLForKeywordSearch(keyword)`.